File: src/estacao_meteorologica.py
from alarme_termico import AlarmeTermico
from sensor_temperatura import SensorTemperatura


class EstacaoMeteorologica:
    def __init__(self, tag: str, temperatura_inicial: float) -> None:
        self._sensor = SensorTemperatura(tag, temperatura_inicial)
        self._alarme = AlarmeTermico()

    def registrar_temperatura(self, temperatura: float) -> bool:
        if not self._sensor.atualizar(temperatura):
            return False

        self._alarme.avaliar(temperatura)

        return True

    @property
    def temperatura(self) -> float:
        return self._sensor.valor

    @property
    def alarme_ligado(self) -> bool:
        return self._alarme.ligado

File: src/estacao_meteorologica.cpp
#include "estacao_meteorologica.hpp"

EstacaoMeteorologica::EstacaoMeteorologica(const std::string& tag, double temperaturaInicial)

    : sensor_(tag, temperaturaInicial) {

}

bool EstacaoMeteorologica::registrarTemperatura(double temperatura) {

    if (!sensor_.atualizar(temperatura)) {
        return false;
    }

    alarme_.avaliar(temperatura);

    return true;
}

double EstacaoMeteorologica::temperatura() const { return sensor_.valor(); }

bool EstacaoMeteorologica::alarmeLigado() const { return alarme_.estaLigado(); }